stackoverflow/src/main/java/ro/utcn/sd/cata/stackoverflow/service/AnswerService.java

```diff
@@ -0,0 +1,51 @@
+package ro.utcn.sd.cata.stackoverflow.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ro.utcn.sd.cata.stackoverflow.entity.Answer;
+import ro.utcn.sd.cata.stackoverflow.entity.Question;
+import ro.utcn.sd.cata.stackoverflow.entity.User;
+import ro.utcn.sd.cata.stackoverflow.exception.AnswerNotFoundException;
+import ro.utcn.sd.cata.stackoverflow.exception.QuestionNotFoundException;
+import ro.utcn.sd.cata.stackoverflow.repository.RepositoryFactory;
+
+import java.time.LocalDateTime;
+
+@Service
+@RequiredArgsConstructor
+public class AnswerService {
+    private final RepositoryFactory repositoryFactory;
+
+    @Transactional
+    public Answer addAnswer(User author, int questionId, String text){
+        return repositoryFactory.createAnswerRepository().save(
+                new Answer(questionId, author.getId(),text, LocalDateTime.now().toString()),repositoryFactory.createQuestionRepository().findById(questionId).orElseThrow(QuestionNotFoundException::new));
+    }
+
+    @Transactional
+    public boolean removeAnswer(int id, int authorId){
+        Answer answerToRemove = repositoryFactory.createAnswerRepository().findById(id).orElseThrow(AnswerNotFoundException::new);
+        if(answerToRemove.getAuthorId().equals(authorId)) {
+            repositoryFactory.createAnswerRepository().remove(answerToRemove);
+            return true;
```

Comment:
Again, here it would be better to throw exceptions instead of having "return codes".